### DISCUSSION TITLE
Solax charge mode change unreliable

### DISCUSSION THREAD

**Discussion by Psai** [wills106/homeassistant-solax-modbus#648] [Q&A]
Hi,

I've looked through various other threads for solutions to the issue I'm facing, but nothing seems to resolve the issue.

(This is a screenshot from a cloned automation with the non-solax stuff removed).

When the auto action fires, sometimes the inverter changes back to "self use" mode after a few seconds.

I've tried adding delays and pressing the unlock button, bit nothing seems to make it reliably work.

![Screenshot_20231129-115425](https://github.com/wills106/homeassistant-solax-modbus/assets/3990521/a2b3cfc5-b5a2-4f2c-a669-603b77c4459e)
![Screenshot_20231129-115433](https://github.com/wills106/homeassistant-solax-modbus/assets/3990521/929d4c97-2041-4cb4-9435-e5a3360d2483)

**Comment by wills106** [comment]
Is the Inverter awake? Or has it gone into idle mode?

If it's in idle mode as no PV and battery at Min SOC, you can't change the mode.
You need to make any changes before the Inverter shuts down.

**Reply by Psai** [reply]
Good to know thanks. This kind of sucks as it means it'll never work as flexibly as I'd have hoped (followed the intelligent octopus charge times)

**Comment by otava5** [comment]
Backup mode could be the solution

**Reply by Psai** [reply]
I'll test this and report back, thanks

**Reply by Psai** [reply]
Ah, unfortunately this starts the battery charging straight away as the backup mode minimum SOC is 30%
